File: practice/38_Count_and_Say.kt
fun main() {
	println(count(4))
}

fun count(n: Int): String {
    return helper1(n, 1, "1")
}

fun helper1(n: Int, start: Int, p: String): String {
	if(n==start) {
		return p
	}
	var count = 1
	var temp = StringBuilder()

	for (i in 0 until p.length) {
		if(p[i] == p[i+1]) {
			count++
		} else {
			temp.append(count).append(p[i])
			count = 1
		}
	}

	return helper1(n, start+1, temp.toString())
}

File: practice/38_Count_and_Say.js
function count(n) {
    return helper(n, 1, '1');
}

function helper(n, start, p) {
    let count = 1,
        temp = '';

    if (start >= n) {
        return p;
    }

    for (let index = 0; index < p.length; index++) {
        if (p[index] === p[index + 1]) {
            count++;
        } else {
            temp += `${count}${p[index]}`;
            count = 1;
        }
    }

    return helper(n, start + 1, temp);
}

console.log(count(4));
